gh-57684: Add the -p command line option (#92361)
python/cpython: Doc/using/cmdline.rst

   * ``python -c code`` and ``python`` (REPL) command lines: Don't prepend an
     empty string, which means the current working directory.

   See also the :envvar:`PYTHONSAFEPATH` environment variable, and :option:`-p`
   and :option:`-I` (isolated) options.

   .. versionadded:: 3.11

.. cmdoption:: -p

>    Prepend a potentially unsafe path to :data:`sys.path`: opposite of the

Reviewer: ```suggestion
   Prepend a potentially unsafe path to :data:`sys.path`; opposite of the
```

A colon doesn't really make syntactic or semantic sense here; a semicolon is the closest match to the existing text, but there are several other ways to rephrase this if desired.

   :option:`-P` option.

   This is the default Python behavior. This option can be used to ignore the
   :option:`-P` command line option and the :envvar:`PYTHONSAFEPATH`
   environment variable.

>    The :option:`-p` option takes precedence over the :option:`-P` and
>    :option:`-I` (isolated mode) options and the :envvar:`PYTHONSAFEPATH`
>    environment variable.

Reviewer: I'd be concerned this might not be entirely clear that this doesn't override all of `-I`, just the `-P` it implies. What about something like this?

```suggestion
   The :option:`-p` option takes precedence over the :option:`-P` option
   (either passed explicitly or implied by :option:`-I`, isolated mode)
   and the :envvar:`PYTHONSAFEPATH` environment variable.
```

.. envvar:: PYTHONSAFEPATH

   If this is set to a non-empty string, don't prepend a potentially unsafe

>    path to :data:`sys.path`: same effect than the :option:`-P` option.

Reviewer: ```suggestion
   path to :data:`sys.path`; same effect as the :option:`-P` option.
```

Fix punctuation (as above) and grammar error
